app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -112,25 +112,50 @@ if [ x$ID = xrhel ] || [ x$ID = xcentos ]; then
 <%= "          --data 'remote_execution_interface=#{@remote_execution_interface}' \\\n" if @remote_execution_interface.present? -%>
 <%= "          --data 'packages=#{@packages}' \\\n" if @packages.present? -%>
 
-}
+    }
 
     <% if @force -%>
-    yum remove -y katello-ca-consumer*
+    yum remove -y katello-ca-consumer* subscription-manager
     <% end -%>
 
-    CONSUMER_RPM=$(mktemp --suffix .rpm)
-    curl --silent --show-error --output $CONSUMER_RPM <%= subscription_manager_configuration_url(hostname: @url_host) %>
+    KATELLO_SERVER_CA_CERT=/etc/rhsm/ca/<%= @url.host %>.pem
+    RHSM_CFG=/etc/rhsm/rhsm.conf
 
-    # Workaround for systems with enabled FIPS,
-    # where installation of RPM generated on RHEL7 cause 'no digest' error
-    # See https://projects.theforeman.org/issues/32068
-    if [ "$(cat /proc/sys/crypto/fips_enabled)" = "1" ]; then
-        rpm -ivh --nodigest --nofiledigest $CONSUMER_RPM
-    else
-        yum localinstall $CONSUMER_RPM -y
+    if ! command -v subscription-manager &> /dev/null
+    then
+      yum install -y subscription-manager
     fi
 
-    rm -f $CONSUMER_RPM
+    if ! [ -f $RHSM_CFG ] ; then
+      echo "'$RHSM_CFG' not found, cannot configure subscription-manager"
+      exit 1
+    fi
+
+    # Prepare SSL certificate
+    cp -f $SSL_CA_CERT $KATELLO_SERVER_CA_CERT
+    chmod 644 $KATELLO_SERVER_CA_CERT
+
+    if [ -d /etc/pki/ca-trust/source/anchors ]; then
+      update-ca-trust enable
+      cp -f $KATELLO_SERVER_CA_CERT /etc/pki/ca-trust/source/anchors
+      update-ca-trust
+    fi
+
+    # Configure subscription-manager
+    test -f $RHSM_CFG.bak || cp $RHSM_CFG $RHSM_CFG.bak
+    subscription-manager config \
+      --server.hostname="<%= @url.host %>" \
+      --server.port="<%= @url.port %>" \
+      --server.prefix="/rhsm" \
+      --rhsm.repo_ca_cert="$KATELLO_SERVER_CA_CERT" \
+      --rhsm.baseurl="<%= @url %>pulp/content"
```

Comment:
I think we still need to include https://github.com/theforeman/puppet-foreman_proxy_content/blob/master/lib/puppet/provider/rhsm_reconfigure_script/rhsm_reconfigure_script.rb#L119-L124